### ISSUE TITLE
Intermittent zip file closed issues

### ISSUE THREAD

**Issue by davidburstromspotify** [google/auto#715]
We're facing intermittent issues while compiling modules with AutoValue using Gradle. The following exception was thrown in a module with com.google.auto.value:auto-value:1.6.3, com.google.auto.factory:auto-factory:1.0-beta5 and com.ryanharter.auto.value:auto-value-parcel:0.2.6. Gradle was running in JDK 1.8.0_151-b12 with 12 workers. Do you have any idea what might be the issue?

```
 Caused by: java.lang.ExceptionInInitializerError
 	at com.google.auto.value.processor.AutoValueProcessor.processType(AutoValueProcessor.java:238)
 	at com.google.auto.value.processor.AutoValueOrOneOfProcessor.process(AutoValueOrOneOfProcessor.java:310)
 	at org.gradle.api.internal.tasks.compile.processing.DelegatingProcessor.process(DelegatingProcessor.java:62)
 	at org.gradle.api.internal.tasks.compile.processing.DynamicProcessor.process(DynamicProcessor.java:52)
 	at org.gradle.api.internal.tasks.compile.processing.DelegatingProcessor.process(DelegatingProcessor.java:62)
 	at org.gradle.api.internal.tasks.compile.processing.TimeTrackingProcessor.access$401(TimeTrackingProcessor.java:37)
 	at org.gradle.api.internal.tasks.compile.processing.TimeTrackingProcessor$5.create(TimeTrackingProcessor.java:99)
 	at org.gradle.api.internal.tasks.compile.processing.TimeTrackingProcessor$5.create(TimeTrackingProcessor.java:96)
 	at org.gradle.api.internal.tasks.compile.processing.TimeTrackingProcessor.track(TimeTrackingProcessor.java:117)
 	at org.gradle.api.internal.tasks.compile.processing.TimeTrackingProcessor.process(TimeTrackingProcessor.java:96)
 	at com.sun.tools.javac.processing.JavacProcessingEnvironment.callProcessor(JavacProcessingEnvironment.java:794)
 	at com.sun.tools.javac.processing.JavacProcessingEnvironment.discoverAndRunProcs(JavacProcessingEnvironment.java:705)
 	at com.sun.tools.javac.processing.JavacProcessingEnvironment.access$1800(JavacProcessingEnvironment.java:91)
 	at com.sun.tools.javac.processing.JavacProcessingEnvironment$Round.run(JavacProcessingEnvironment.java:1035)
 	at com.sun.tools.javac.processing.JavacProcessingEnvironment.doProcessing(JavacProcessingEnvironment.java:1176)
 	at com.sun.tools.javac.main.JavaCompiler.processAnnotations(JavaCompiler.java:1170)
 	at com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:856)
 	at com.sun.tools.javac.main.Main.compile(Main.java:523)
 	... 104 more
 Caused by: java.lang.IllegalStateException: zip file closed
 	at com.google.auto.value.processor.TemplateVars.readerFromResource(TemplateVars.java:139)
 	at autovalue.shaded.com.google$.escapevelocity.$Parser.parseParse($Parser.java:427)
 	at autovalue.shaded.com.google$.escapevelocity.$Parser.parseDirective($Parser.java:331)
 	at autovalue.shaded.com.google$.escapevelocity.$Parser.parseNode($Parser.java:218)
 	at autovalue.shaded.com.google$.escapevelocity.$Parser.parseTokens($Parser.java:126)
 	at autovalue.shaded.com.google$.escapevelocity.$Parser.parse($Parser.java:118)
 	at autovalue.shaded.com.google$.escapevelocity.$Template.parseFrom($Template.java:101)
 	at com.google.auto.value.processor.TemplateVars.parsedTemplateForResource(TemplateVars.java:119)
 	at com.google.auto.value.processor.AutoValueTemplateVars.<clinit>(AutoValueTemplateVars.java:145)
 	... 122 more

```

**Comment by eamonnmcmanus** [comment]
Yes, I know exactly what the issue is. We already have a [workaround](https://github.com/google/auto/blob/master/value/src/main/java/com/google/auto/value/processor/TemplateVars.java#L122) for this issue, which is a [JDK bug](https://bugs.openjdk.java.net/browse/JDK-6947916). However, I see that we don't catch `IllegalStateException`, so the workaround doesn't get triggered for your case. You could try a more recently JDK 8 release, but I suspect that the fix hasn't been backported into any publicly-available release. Or you could try building on JDK 9+.

I was planning to make a new AutoValue release soon, so I'll make sure it includes the obvious fix (adding `IllegalStateException` on the line I linked to above).

**Comment by davidburstromspotify** [comment]
Ah, thank you. Indeed the fix has been released in 1.8.0_161, which is available. Sorry to bother you.
